packages/firebase_auth/lib/src/firebase_auth.dart

```diff
@@ -126,14 +116,10 @@ class FirebaseAuth {
     @required String email,
   }) async {
     assert(email != null);
-    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-    // https://github.com/flutter/flutter/issues/26431
-    // ignore: strong_mode_implicit_dynamic_method
-    final List<dynamic> providers = await channel.invokeMethod(
+    return await channel.invokeListMethod<String>(
```

Comment:
Wouldn't [invokeListMethod](https://docs.flutter.io/flutter/services/MethodChannel/invokeListMethod.html) crash when `result.cast<T>();` is called and `providers == null`.

This makes me wonder if we should use `invokeListMethod` and `invokeMapMethod` in this PR because in some cases we want to return null values.